lib/rspec/matchers/built_in/yield.rb

```diff
@@ -369,16 +363,18 @@ def initialize(*args)
         # @private
         def matches?(block)
           @actual_formatted = []
+          @actual = []
           args_matched_when_yielded = true
           yield_count = 0
 
-          @probe = YieldProbe.probe(block) do |args|
-            @actual_formatted << RSpec::Support::ObjectFormatter.format(args)
-            args_matched_when_yielded &&= values_match?(@expected[yield_count], args)
+          @probe = YieldProbe.probe(block) do |*arg_array|
```

Comment:
is adding the splat the key? Not immediately clear to me what in this change actually fixes it. (Inlining `successive_yield_args` seems good overall regardless though.)